azext_iot/_params.py

```diff
@@ -772,6 +782,48 @@ def load_arguments(self, _):
             help="Maximum number of configurations to return. By default all configurations are returned.",
         )
 
+    with self.argument_context("iot hub state migrate") as context:
+        context.argument(
+            "hub_name",
+            options_list=["--destination-hub", "--dh"],
+            help="Name of IoT Hub to which the origin hub will be copied."
+        )
+        context.argument(
+            "resource_group_name",
+            options_list=["--destination-resource-group", "--dg"],
+            help="Name of resource group of the IoT Hub to which the origin hub will be copied."
+        )
+        context.argument(
+            "login",
+            options_list=["--destination-hub-login", "--dl"],
+            validator=mode2_iot_login_handler,
+            help="This command supports an entity connection string with rights to perform action on the destination hub. "
+            'Use to avoid session login via "az login" for this IoT Hub instance. '
+            "If both an entity connection string and name are provided the connection string takes priority. "
+            "Required if --hub-name is not provided.",
+            arg_group="IoT Hub Identifier"
+        )
+        context.argument(
+            "orig_hub",
+            options_list=["--origin-hub", "--oh"],
+            help="Name of IoT Hub which will be copied."
+        )
+        context.argument(
+            "orig_resource_group_name",
+            options_list=["--origin-resource-group", "--og"],
+            help="Name of resource group of the IoT Hub which will be copied."
+        )
+        context.argument(
+            "orig_hub_login",
+            options_list=["--origin-hub-login", "--ol"],
+            validator=mode2_iot_login_handler,
+            help="This command supports an entity connection string with rights to perform action on the origin hub. "
+            'Use to avoid session login via "az login" for this IoT Hub instance. '
+            "If both an entity connection string and name are provided the connection string takes priority. "
+            "Required if --hub-name is not provided.",
```

Comment:
required if what is not provided?

Also not sure if this should be underneath a grouping since the origin hub isnt under it